code/go/0chain.net/blobbercore/datastore/postgres.go

```diff
@@ -97,8 +103,33 @@ func (store *postgresStore) GetTransaction(ctx context.Context) *EnhancedDB {
 	return nil
 }
 
-func (store *postgresStore) WithTransaction(ctx context.Context, tx *gorm.DB) context.Context {
-	return context.WithValue(ctx, ContextKeyTransaction, EnhanceDB(tx))
+func (store *postgresStore) WithNewTransaction(f func(ctx context.Context) error) error {
+	ctx := store.CreateTransaction(context.TODO())
+	defer ctx.Done()
+
+	tx := store.GetTransaction(ctx)
+	err := f(ctx)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	tx.Commit()
+	return nil
+}
+func (store *postgresStore) WithTransaction(ctx context.Context, f func(ctx context.Context) error) error {
```

Comment:
I do not see its usage anywhere? Is this method on the `store` is required anymore or not?

Reply:
we need it, it will be used eventually